packages/flutter/lib/src/cupertino/multicolumn_picker.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```

Comment:
I like the idea of making the pickers composable. But this potentially create a layout challenge. Imagine 2 columns with the same width, then their offAxisFractions are the same and that case is easy. Now if column 2 has one choice in there that's super long, then now column one is pushed further to the left to keep everything centered. But then visually, it means column one's offAxisFraction should be bigger than column two now but column one can't know what choices are inside column two. 

I would test some of these extremities to make sure it doesn't look visually too strange.

Reply:
When creating a multi column picker, I think users can calculate beforehand how much "offAxis" they want each column to be, to give the correct visual effect, and then pass the parameter down to each CupertinoPicker. So each picker itself can't know about other pickers' width and position, but it has been given a parameter to calculate its center. Therefore I don't think there's a problem with this.

Reply:
For the sake of simplicity, I suspect we may need to come back and add in logic to calculate the correct center point for each cylinder.

This will likely need to be done as part of setting up the DatePicker widget later on.